nav2_costmap_2d/include/nav2_costmap_2d/array_parser.h

```diff
@@ -28,8 +28,8 @@
  *
  * author: Dave Hershberger
  */
-#ifndef nav2_costmap_2d_ARRAY_PARSER_H
-#define nav2_costmap_2d_ARRAY_PARSER_H
+#ifndef NAV2_COSTMAP_2D_ARRAY_PARSER_H
```

Comment:
Does this pass cpplint? I thought it wanted two _ between the path and the filename part eg `NAV2_COSTMAP_2D__ARRAY_PARSER_H`

Reply:
This directory has a lot of issues still w/ the linters. Since the spirit of this PR was directory re-org, I suggest leaving that for a separate PR.

Reply:
fixed all to have double underscores.